File: gitlfs-common/src/main/java/ru/bozaro/gitlfs/common/data/Ref.java
package ru.bozaro.gitlfs.common.data;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.annotation.CheckForNull;
import javax.annotation.Nonnull;

public final class Ref {

  /**
   * Fully-qualified server refspec.
   */
  @JsonProperty(value = "name", required = true)
  @Nonnull
  private final String name;

  public Ref(@JsonProperty(value = "name", required = true) @Nonnull String name) {
    this.name = name;
  }

  @CheckForNull
  public static Ref create(@CheckForNull String ref) {
    return ref == null ? null : new Ref(ref);
  }

  @Nonnull
  public String getName() {
    return name;
  }

  @Override
  public String toString() {
    return name;
  }
}

File: gitlfs-common/src/main/kotlin/ru/bozaro/gitlfs/common/data/Ref.kt
package ru.bozaro.gitlfs.common.data

import com.fasterxml.jackson.annotation.JsonProperty

class Ref(
        /**
         * Fully-qualified server refspec.
         */
        @field:JsonProperty(value = "name", required = true)
        @param:JsonProperty(value = "name", required = true)
        val name: String) {
    override fun toString(): String {
        return name
    }

    companion object {
        fun create(ref: String?): Ref? {
            return if (ref == null) null else Ref(ref)
        }
    }
}
